### PR TITLE
[ENG-18] Add request context middleware to orka

### PR DESCRIPTION
**Pull request by klesgidis**
## Summary
This is the first PR of adding a `Request Context` to orka.

Example for how async_hooks work: https://www.icloud.com/keynote/0uwcgADQfQRGhUDkzAZ4SMd7A#AsyncLocalStorage

This change requires NodeJS: 12.17.0+, 13.14.0+, or 14.0.0+

Another PR will follow:
* Reading this context and exposing those variables to the log4js appenders
* Adding an exampe in the `/examples` folder

### CODE REVIEW

**Review by nikosd23 — COMMENTED** [review]
🥇

File: src/builder.ts [hunk 1]
```diff
@@ -4,6 +4,10 @@ import _defaults from './default-options';
 import * as lodash from 'lodash';
 import { getLogger } from './initializers/log4js';
 import orkaType from './orka-builder';
+import { AsyncLocalStorage } from 'async_hooks';
```
Comment (nikostoulas) [inline src/builder.ts:7]: ❓ Will this break if used with node < 12.17 ?

File: src/initializers/koa/add-visitor-id.ts [hunk 2]
```diff
@@ -14,8 +15,9 @@ export default function(config): Middleware {
     try {
       const { cookie_id: visitor } = decode(cookie);
       ctx.state.visitor = visitor;
+      getRequestContext()?.set('visitor', ctx.state.visitor);
```
Comment (nikostoulas) [inline src/initializers/koa/add-visitor-id.ts:18]: 👏

**Review by nikostoulas — APPROVED** [review]
Nicely done @klesgidis! I was wondering if this is a breaking change (and should bump our major version) or if there is a way to make it non breaking.

File: src/builder.ts (hunk 1 above)
Comment (klesgidis) [reply src/builder.ts:7]: Yes this will break if node < 12.17

We could go for `v1.0.0`

Comment (klesgidis) [reply src/builder.ts:7]: I could also enhance readme with the minimum NodeJS version

**Review by echetzakis — APPROVED** [review]
👌  nice!

**Pull request merged**